### PR TITLE
fix: Metrics OverWrite issue on pushgateway

### PR DESCRIPTION
**Pull request by sarabala1979**

### CODE REVIEW

File: prometheus-pusher/main.go [hunk 1]
```diff
@@ -65,10 +65,14 @@ func (p *prometheusSink) push(msgPayloads []Payload) error {
 		case "Gauge":
 			p.logger.Debugf("Creating Collector %s", payload.Name)
 			pusher = pusher.Collector(&myCollector{
-				metric:     prometheus.NewDesc(payload.Name, "", nil, payload.Labels),
+				metric:     prometheus.NewDesc(payload.Name, "", nil, nil),
 				metricType: prometheus.GaugeValue,
 				value:      payload.Value,
 			})
+
+			for key, value := range payload.Labels {
```
Comment (sarabala1979) [inline prometheus-pusher/main.go:73]: Pushgateway will convert the grouping key into label on /metrics

Comment (sarabala1979) [reply prometheus-pusher/main.go:73]:
```
<img width="1541" alt="image" src="https://user-images.githubusercontent.com/33908564/228697211-110d347f-40cf-42b7-9e8d-9afe1c483092.png">

Comment (sarabala1979) [reply prometheus-pusher/main.go:73]: tested with standalone push gateway

File: prometheus-pusher/main_test.go [hunk 2]
```diff
@@ -46,3 +55,56 @@ func TestMergeLabel(t *testing.T) {
 	assert.Len(t, pl.Labels, 3)
 
 }
+
+func TestPusher(t *testing.T) {
+	// Fake a Pushgateway that responds with 202 to DELETE and with 200 in
+	// all other cases.
```
Comment (sarabala1979) [inline prometheus-pusher/main_test.go:61]: This test case just make sure Prometheus push is pushing the metrics in right format

**Review by ab93 — APPROVED** [review]

**Review by vigith — APPROVED** [review]

**Pull request merged**